policyengine_api/endpoints/search/search.py

```diff
@@ -1,103 +1,167 @@
+# )
 
-if not (folder / "metadata.csv.gz").exists():
-    METADATA_URL = "https://api.github.com/repos/PolicyEngine/policyengine-api/releases/assets/103041106"
-    METADATA_PATH = folder / "metadata.csv.gz"
-    METADATA_PATH.write_bytes(
-        requests.get(
-            METADATA_URL, headers={"Accept": "application/octet-stream"}
-        ).content
-    )
+# with h5py.File("embeddings.h5", "r") as f:
+#     embeddings = f["embeddings"][:]
 
-metadata_df = pd.read_csv(folder / "metadata.csv.gz", compression="gzip")
-folder = Path(".")
+# metadata_df["embedding"] = embeddings.tolist()
 
-metadata_df[["name", "country_id", "type"]].to_csv(
-    "metadata.csv.gz", index=False, compression="gzip"
-)
+# embedding_dimensions = len(embeddings[0])
 
-with h5py.File("embeddings.h5", "r") as f:
-    embeddings = f["embeddings"][:]
+# # Use FAISS to create an index
 
-metadata_df["embedding"] = embeddings.tolist()
+# import faiss
 
-embedding_dimensions = len(embeddings[0])
+# index_names = [
+#     "uk_parameters",
+#     "uk_variables",
+#     "us_parameters",
+#     "us_variables",
+# ]
 
-# Use FAISS to create an index
+# indexes = {
+#     name: faiss.IndexFlatL2(embedding_dimensions) for name in index_names
+# }
 
-import faiss
+# names = {name: [] for name in index_names}
 
-index_names = [
-    "uk_parameters",
-    "uk_variables",
-    "us_parameters",
-    "us_variables",
-]
+# for metadata_type in ["parameter", "variable"]:
+#     for country_id in COUNTRIES:
+#         name = f"{country_id}_{metadata_type}s"
+#         index = indexes[name]
+#         metadata = metadata_df[
+#             (metadata_df["type"] == metadata_type)
+#             * (metadata_df["country_id"] == country_id)
+#         ]
+#         embeddings = np.array(metadata["embedding"].tolist()).reshape(
+#             -1, embedding_dimensions
+#         )
+#         index.add(embeddings)
+#         names[name] = metadata["name"].tolist()
 
-indexes = {
-    name: faiss.IndexFlatL2(embedding_dimensions) for name in index_names
-}
 
-names = {name: [] for name in index_names}
+def get_search():
+    """
+    get_search (Removed)
+    ---
+    This endpoint has been removed and is no longer functional.
 
-for metadata_type in ["parameter", "variable"]:
-    for country_id in COUNTRIES:
-        name = f"{country_id}_{metadata_type}s"
-        index = indexes[name]
-        metadata = metadata_df[
-            (metadata_df["type"] == metadata_type)
-            * (metadata_df["country_id"] == country_id)
-        ]
-        embeddings = np.array(metadata["embedding"].tolist()).reshape(
-            -1, embedding_dimensions
-        )
-        index.add(embeddings)
-        names[name] = metadata["name"].tolist()
+    Description:
+        This endpoint previously searched using hugging face embeddings and torch.
+        It has been removed due to infrequent usage and performance considerations.
 
+    HTTP Method:
+        GET
+
+    URL:
+        /country_id/search
+
```

Comment:
Ah, I now see what you meant in our meeting today, @tawandamoyo.

First off, I think you're fine to delete all of the old code and reference to the endpoint in the Open API spec. This was really only used internally for experimental purposes, so I don't think we even need a reference to it here.

Second, I think your documentation structure here is very good, but more suited to docs _outside_ of the code itself. This is a structure you'd see in a published docs website or GitHub Pages site, for example.

At this point in the code, which is where our controller (get_search) is positioned, I would use a docstring to define the following:
* Any values that might be passed via a URL directory (e.g., `country_id`), which you'll find in the `api.py` file, where the routes are
* Any values that will be pulled from URL parameters, such as query parameters (e.g., "?value=something"). These need to be taken from the request object itself, via `request.args.get`
* Any required HTTP headers, and what should happen if they're not included (e.g., if we needed a particular `Content-Type` or `Authentication` header

Otherwise, I would put all of the information you've provided here in an external doc, but honestly, I'd structure it quite similarly.